ddd-koans/Value Object/Immutablity/test/Tests.kt

```diff
@@ -0,0 +1,33 @@
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Test
+import kotlin.reflect.KMutableProperty
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.jvmErasure
+
+class Test {
+    @Test
+    fun testIfTimeSignatureIsImmutable() {
+        TimeSignature::class.declaredMemberProperties.forEach {
+            assertFalse(it is KMutableProperty<*>, "${it.name} should be declared as val not var")
+            assertEquals(Int::class, it.returnType.jvmErasure, "${it.name} should be declared with type Int")
+            assertFalse(it.returnType.isMarkedNullable, "${it.name} should not be marked as nullable")
+        }
+    }
+
+    @Test
+    fun testIfTimeSignatureHasOnlyTwoAttributesOfDenominatorAndNumerator() {
+        val expectedPropertiesNames = setOf("numerator", "denominator")
+
+        val declaredMemberProperties = TimeSignature::class.declaredMemberProperties
+        val declaredPropertiesNames = declaredMemberProperties.map { it.name }.toSet()
+
+        assertEquals(2, declaredMemberProperties.size, "TimeSignature should has exactly two properties")
+        assertEquals(
```

Comment:
is it's possible to use:
1. at least different matches like assert4j
2. use Kotest instead of Junit maybe?

Reply:
I'll test this options. However, if we would like to port the code to Java course one day, JUnit is probably better than Kotest. I can also try assert4j. 

I would love to answer questions with 100 % certainty, however, the documentation is not the best in the world right now.

Reply:
you're right with JUnit, so maybe just more expressive assertions will be enough